geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/io/reader/StreamingSimpleFeatureArrowFileReader.scala

```diff
@@ -0,0 +1,143 @@
+/***********************************************************************
+* Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.arrow.io.reader
+
+import java.io.{Closeable, InputStream}
+
+import org.apache.arrow.memory.BufferAllocator
+import org.apache.arrow.vector.complex.NullableMapVector
+import org.apache.arrow.vector.stream.ArrowStreamReader
+import org.locationtech.geomesa.arrow.features.ArrowSimpleFeature
+import org.locationtech.geomesa.arrow.io.SimpleFeatureArrowFileReader.{SkipIndicator, VectorToIterator}
+import org.locationtech.geomesa.arrow.io.{SimpleFeatureArrowFileReader, SimpleFeatureArrowIO}
+import org.locationtech.geomesa.arrow.vector.{ArrowDictionary, SimpleFeatureVector}
+import org.opengis.feature.simple.SimpleFeatureType
+import org.opengis.filter.Filter
+
+import scala.collection.mutable.ArrayBuffer
+
+/**
+  * Streams features from an input stream - note that a feature may not be valid after a call to `.next()`,
+  * as the underlying data may be reclaimed
+  *
+  * @param is creates a new input stream
+  * @param allocator buffer allocator
+  */
+class StreamingSimpleFeatureArrowFileReader(is: () => InputStream)(implicit allocator: BufferAllocator)
```

Comment:
Would it be possible to take an InputStream directly so that we needn't read the metadata separate from the rest of the file read?